NYPL/cantaloupe
# DR-2706, DR-2707 purge cache script

This adds a script for purging the cache (DR-2706), and a file with some items that need purging in production (DR-2707, Maps of North America).

**To test**

Copy the four new environment variables from .env.example to .env. You can use the default values:

```
ENDPOINT_API_USERNAME=admin
ENDPOINT_API_SECRET=secret
ENDPOINT_ADMIN_USERNAME=admin
ENDPOINT_ADMIN_SECRET=secret
```

Rebuild and start your local cantaloupe app:

`docker-compose build && docker-compose up`

Run the script with the -f flag to purge the identifiers from DR-2707 from your local cache:

`ruby scripts/purge_cache.rb -f scripts/DR-2707-purge-maps-of-north-america-cache.txt`

Very you see a series of "PurgeItemFromCache" events in the admin console (only the last 10 will be shown):

`http://localhost:8182/admin`

Run the script with the -A flag to purge all identifiers from your local cache:

`ruby scripts/purge_cache.rb -A`
